### PR TITLE
Remove Bazel4 specific hacks

### PR DESCRIPTION
**Pull request by jerrymarino**
We needed several changes to enable Apple silicon on Bazel 4 that are
obselete with Bazel 5.

Drop toolchains, rules_swift fork, features. This adds a lot of
complexity and is error prone to keep around.

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -29,23 +29,8 @@ build --features swift.index_while_building
...
-common:bazel4_arm64_simulator_arm64_host --features bazel4.override_host_cpu_arm64
-common:bazel4_arm64_simulator_arm64_host --features bazel4.override_simulator_cpu_arm64
-
-# Similar to above but doesn't override the host CPU
-common:bazel4_arm64_simulator_x86_64_host --features apple.arm64_simulator_use_device_deps
-common:bazel4_arm64_simulator_x86_64_host --features bazel4.override_simulator_cpu_arm64
-
-# Broken on Intel + Bazel 5 so turnoff
-build --deleted_packages tools/toolchains/bazel4_local_config_cc
+# Flip this on 100% of time - conditiionally enabled
```
Comment (dierksen) [inline .bazelrc:32]:
```suggestion
# Flip this on 100% of time - conditionally enabled
```

**Review by thiagohmcruz — APPROVED** [review]
🎉

**Comment by dierksen** [comment]
Looks like a fail on https://github.com/bazel-ios/rules_ios/blob/052ecc1544e72aeed35a722235f7b657478f6072/rules/import_middleman.bzl#L174 being masked by the `lablel` on that line.

**Comment by jerrymarino** [comment]
> https://github.com/bazel-ios/rules_ios/pull/410#issuecomment-1190729604

Ah good catch! Yeah I ended up removing this in a followup - thanks for the review 🚀

**Pull request merged**